adagrams/game.py

```diff
@@ -1,13 +1,129 @@
 from random import randint
+from .constants import LETTER_POOL, SCORE_CHART, ALPHABET
 
 def draw_letters():
-    pass
+
+    alphabet_pool = dict(LETTER_POOL)
+    results = []
+
+    while len(results) < 10:
+
+        random_num = randint(0, 25)
+
+        letter = ALPHABET[random_num]
+
+        if alphabet_pool[letter] > 0:
+            results.append(letter)
+            alphabet_pool[letter] -= 1
+
+    return results
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    
+    cap_word = word.upper()
+    letter_tracker = create_letter_tracker(letter_bank)
+    result = True
+
+    for letter in cap_word:
+        if letter in letter_tracker and letter_tracker[letter] > 0:
+            letter_tracker[letter] -= 1
+        else:
+            result = False
+            break
```

Comment:
I would reverse the sense of this comparison so that we can unindent the logic a little.

Recall that we should always prefer to have our main logic unindented as much as possible.

This means that you should check if a letter is not available and immediately return False and then the main logic of decrementing the value of a letter is unindented and not nested in conditional logic.